File: src/dotnet/Core/Domain/Entities/RankedCandidate.cs
namespace Semantic.Kernel.Api.Core.Domain.Entities;

public sealed record RankedCandidate
{
    public required string CandidateId { get; init; }
    public required List<string> Documents { get; init; }
    public required Dictionary<string, object> Metadata { get; init; }
    public required double MaxScore { get; init; }
    public required List<double> AllScores { get; init; }
    public required double TechnicalScore { get; init; }
    public required double SeniorityScore { get; init; }
    public required double LeadershipScore { get; init; }
    public required double ExperienceScore { get; init; }
    public required double TotalScore { get; init; }
}

File: src/python/core/domain/entities/ranked_candidate.py
from dataclasses import dataclass
from typing import List, Dict, Any


@dataclass
class RankedCandidate:
    candidate_id: str
    documents: List[str]
    metadata: Dict[str, Any]
    max_score: float
    all_scores: List[float]
    technical_score: float
    seniority_score: float
    leadership_score: float
    experience_score: float
    total_score: float
